src/Corax/Queries/SortingMatch/SortingMatch.cs

```diff
@@ -251,9 +257,13 @@ private struct Fetcher : IFetcher
                     items = MemoryMarshal.Cast<byte, MatchComparer<TComparer, TOut>.Item>(new Span<byte>(match._buffer, match._bufferSize));
                 }
 
+                ref var itemsRef = ref MemoryMarshal.GetReference(items);
+                ref var itemsLast = ref items[^1];
+
+                MatchComparer<TComparer, TOut>.Item cur = default;
                 for (int i = 0; i < read; i++)
                 {
-                    var cur = new MatchComparer<TComparer, TOut>.Item { Key = matches[i], };
+                    cur.Key = matches[i];
```

Comment:
matchesRef here, to? To avoid the check?

Reply:
Yes